src/make_trade.ts

```diff
@@ -434,12 +451,39 @@ async function waitForTrade(
   // for the executed sell amount before concluding no trade happened.
   const sawTradeEvent = await Promise.race([traded, timeout]);
   if (!sawTradeEvent) {
-    return !(await api.getExecutedSellAmount(uid)).isZero();
+    return !(await api.getExecutedSellAmount({ uid })).isZero();
   } else {
     return true;
   }
 }
 
-function formatAmount(amount: BigNumber, { decimals }: TokenInfo): string {
+function formatAmount(amount: BigNumberish, { decimals }: TokenInfo): string {
   return ethers.utils.formatUnits(amount, decimals);
 }
+
+function validTo(): number {
+  // getTime returns milliseconds, we are looking for seconds
+  const now = Math.floor(new Date().getTime() / 1000);
+  // valid 15 minutes
+  return now + 900;
+}
+
+async function getQuote(
+  api: Api,
+  sellToken: string,
+  buyToken: string,
+  sellAmountBeforeFee: BigNumberish,
+  from: string
+): Promise<Order> {
+  const { quote } = await api.getQuote({
+    sellToken,
+    buyToken,
+    validTo: validTo(),
```

Comment:
I see that this parameter is required by the API, but why is it? It seems weird that we need to make up a `validTo` on the spot.

Reply:
You do need it when you place the order and since all parameters of the order may influence your fee (e.g. we may charge you more or less if the validTo date is far in the future) it is required already on the quote.

Reply:
We basically wanted to "future-proof" the quote API by sending every parameter that is part of order creation (other than the amounts that the API derives for you `{sell,buy}Amount` and `feeAmount`).